File: json/json-annotation-processor/src/test/java/ru/tinkoff/kora/json/annotation/processor/dto/DtoWithSupportedTypes.java
package ru.tinkoff.kora.json.annotation.processor.dto;

import ru.tinkoff.kora.json.common.annotation.Json;

import java.math.BigInteger;
import java.util.List;
import java.util.Set;

@Json
public record DtoWithSupportedTypes(
    String string,
    Boolean booleanObject,
    boolean booleanPrimitive,
    Integer integerObject,
    int integerPrimitive,
    BigInteger bigInteger,
    Double doubleObject,
    double doublePrimitive,
    Float floatObject,
    float floatPrimitive,
    Long longObject,
    long longPrimitive,
    Short shortObject,
    short shortPrimitive,
    byte[] binary,
    List<Integer> listOfInteger,
    Set<Integer> setOfInteger) {

}

File: json/json-symbol-processor/src/test/kotlin/ru/tinkoff/kora/json/ksp/dto/DtoWithSupportedTypes.kt
package ru.tinkoff.kora.json.ksp.dto

import ru.tinkoff.kora.json.common.annotation.Json
import java.math.BigInteger

@Json
data class DtoWithSupportedTypes(
    val string: String?,
    val boolean: Boolean,
    val integer: Int,
    val bigInteger: BigInteger,
    val double: Double,
    val float: Float,
    val long: Long,
    val short: Short,
    val binary: ByteArray,
    val listOfInteger: List<Int>,
    val setOfInteger: Set<Int>
)
